my/time/tz/via_location.py

```diff
@@ -46,20 +53,43 @@ class DayWithZone(NamedTuple):
     zone: Zone
 
 
-def _iter_local_dates(start=0, stop=None) -> Iterator[DayWithZone]:
-    finder = _timezone_finder(fast=_FASTER) # rely on the default
+from my.location.common import LatLon
+
+# for backwards compatibility
+def _locations() -> Iterator[Tuple[LatLon, datetime]]:
+    try:
+        import my.location.all
+        for loc in my.location.all.locations():
+            if loc.accuracy is not None and loc.accuracy > config.require_accuracy:
+                continue
+            yield ((loc.lat, loc.lon), loc.dt)
+
+    except Exception as e:
+        from my.core.warnings import high
+        logger.exception("Could not setup via_location using my.location.all provider, falling back to legacy google implemetation", exc_info=e)
+        high("Setup my.google.takeout.parser, then my.location.all for better google takeout/location data")
+
+        import my.location.google
+
+        for gloc in my.location.google.locations():
+            yield ((gloc.lat, gloc.lon), gloc.dt)
+
+
+# Note: since _locations isn't sorted, this is not sorted either
+def _iter_local_dates() -> Iterator[DayWithZone]:
+    finder = _timezone_finder(fast=config.fast) # rely on the default
     pdt = None
     warnings = []
     # todo allow to skip if not noo many errors in row?
-    for l in locations(start=start, stop=stop):
+    for (lat, lon), dt in _locations():
         # TODO right. its _very_ slow...
-        zone = finder.timezone_at(lng=l.lon, lat=l.lat)
+        zone = finder.timezone_at(lat=lat, lng=lon)
         if zone is None:
-            warnings.append(f"Couldn't figure out tz for {l}")
+            warnings.append(f"Couldn't figure out tz for {lat}, {lon}")
             continue
         tz = pytz.timezone(zone)
         # TODO this is probably a bit expensive... test & benchmark
-        ldt = l.dt.astimezone(tz)
+        ldt = dt.astimezone(tz)
         ndate = ldt.date()
```

Comment:
I think this section may need to be removed? Since we cant control the order of my.location

So, this will be slower, but thats the trade off for allowing any sort of location data as input

Its not trivial to cache the `my.location.all` function either, since it depends on any amount of inputs, so creating an accuracte cachew depends_on would be difficult

Reply:
Hmm yeah, I mean in principle we could either demand the upstream providers to provide it as sorted (and for many it'll already be, like google takeout?).
Or alternatively sort before merging them in `all` (via a helper `merge` function), then the whole thing would be much more iterative. But I guess can think about it later

Reply:
Or could even do some magic, e.g. function which takes multiple generator factories (functions `() -> Iterator`), assumes that they are emitting sorted values, and tries merging with https://docs.python.org/3/library/heapq.html#heapq.merge 

If something out of order is detected, it starts from scratch with the offending factory wrapped in `sorted` (hence merging factories, not plain `Iterator`s). I guess realistically it would very quickly lead to detecting which iterables are already sorted and which aren't, usually the order is either increasing, decreasing or just completely random? 

Just leaving it here, but potentially might be useful for other data sources too, e.g. often it's kinda nice to get things according to timestamps

Reply:
Yeah, use a heapmerge would make sense in situations like these, but I still think it might be prone to errors because a user might just overwrite `all.py` with something else (like I've done [here](https://github.com/seanbreckenridge/HPI-personal/blob/master/my/location/all.py)) and then the magic doesn't work anymore. As usual, the magic might make it nice for us, but its not super extendible -- I feel like changing an `all.py` file shouldnt break a separate module

For now I think I'll go with the iterative approach of sorting the `_locations` in `_iter_tzs` -- since its cached when its used repeatedly

Reply:
Another idea could be in `all.py` modules to specify the sources as a `SOURCES = [ ]` key at the toplevel, so then we could actually do some sort of magic and sort the iterators by querying that, instead of doing it in `all.py` itself. Otherwise, `my.location.all.locations()` is just a function and no way to get what the underlying iterators are. If the `SOURCES` var doesn't exist, then fallback to sorting the giant list of locations iteratively

And then locations in `my.location.all` would just be:

```
for provider in SOURCES:
    yield from provider()
```